storage/merkle/tree_test.go

```diff
@@ -338,6 +338,14 @@ func TestRandomOrder(t *testing.T) {
 	rand.Shuffle(len(keys), func(i int, j int) {
 		keys[i], keys[j] = keys[j], keys[i]
 	})
+
```

Comment:
Not in this spot, but what do you think of adding a separate test for proofs with non-existent keys, including proofs for empty tree?

Reply:
yup, I'm going to add several tests as soon as the proof structure is approved.